### PR TITLE
Add sai recorded log player

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: player/player.cpp [hunk 1]
```diff
@@ -0,0 +1,1144 @@
...
+{
+    on_switch_state_change,
+        on_fdb_event,
+        on_port_state_change,
+        on_port_event,
+        on_switch_shutdown_request,
+        on_packet_event
+};
+
+#define EXIT_ON_ERROR(x)\
```
Comment (qiluo-msft) [inline player/player.cpp]: Move this function to a header? I met it in previous code review.

File: player/player.cpp [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+{\
+    sai_status_t s = (x);\
+    if (s != SAI_STATUS_SUCCESS)\
+    {\
+        SWSS_LOG_ERROR("fail status %d", s);\
+        exit(EXIT_FAILURE);\
+    }\
+}
+
+std::vector<std::string> split(const std::string &s, char delim = ',')
```
Comment (qiluo-msft) [inline player/player.cpp]: Same as https://github.com/Azure/sonic-swss-common/blob/master/common/tokenize.cpp?

File: player/player.cpp [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+        sai_attr_serialization_type_t serialization_type;
+        sai_status_t status = sai_get_serialization_type(object_type, attr.id, serialization_type);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("unable to find serialization type for object type %x, attribute %x", object_type, attr.id);
+            exit(EXIT_FAILURE);
+        }
+
+#define CHECK_LIST(x)\
```
Comment (qiluo-msft) [inline player/player.cpp]: Define macro function outside any function?

File: player/player.cpp [hunk 4]
```diff
@@ -0,0 +1,1144 @@
...
+    }
+
+    if (oid != redis_to_local[get_oid])
+    {
+        SWSS_LOG_ERROR("match failed, oid order is mismatch :( oid 0x%lx get_oid 0x%lx second 0x%lx",
+                oid,
+                get_oid,
+                redis_to_local[get_oid]);
+
+        exit(1);
```
Comment (qiluo-msft) [inline player/player.cpp]: magic number?

File: player/player.cpp [hunk 5]
```diff
@@ -0,0 +1,1144 @@
...
+            handle_get_response(object_type, attr_count, attr_list, response);
+        }
+    }
+
+    infile.close();
+
+    return 0;
+}
+
+int main(int argc, char **argv)
```
Comment (qiluo-msft) [inline player/player.cpp]: suggest to add a 'help' parameter and output usage.

File: player/player.cpp (hunk 1 above)
Comment (kcudnik) [inline player/player.cpp]: there was return_on_error if i remember which does different thing

File: player/player.cpp (hunk 3 above)
Comment (kcudnik) [inline player/player.cpp]: fixed

File: player/player.cpp (hunk 4 above)
Comment (kcudnik) [inline player/player.cpp]: fixed

File: player/player.cpp (hunk 5 above)
Comment (kcudnik) [inline player/player.cpp]: added

File: player/player.cpp (hunk 2 above)
Comment (qiluo-msft) [inline player/player.cpp]: Closed.

**Comment by qiluo-msft** [comment]
:clock1:

File: player/player.cpp (hunk 1 above)
Comment (qiluo-msft) [inline player/player.cpp]: So it makes sense to collect similar things together.

Comment (qiluo-msft) [inline player/player.cpp]: Ignore this comment if you cannot find it.

**Pull request merged**